vic/drivers/cesm/cpl_esmf/lnd_comp_mct.F90

```diff
@@ -0,0 +1,224 @@
+
+! !INPUT/OUTPUT PARAMETERS:
+
+    type(ESMF_Clock)            , intent(inout) :: EClock
+    type(seq_cdata)             , intent(inout) :: cdata
+    type(mct_aVect)             , intent(inout) :: x2d, d2x
+    type(seq_cdata)             , intent(inout) :: cdata_s
+    type(mct_aVect)             , intent(inout) :: x2s, s2x
+    character(len=*), optional  , intent(in)    :: NLFilename ! Namelist filename
+
+!EOP
+
+    !--- local variables ---
+    integer(IN)                           :: mpicom
+    integer(IN)                           :: COMPID
+    type(mct_gsMap)             , pointer :: gsMap
+    type(mct_gGrid)             , pointer :: dom
+    type(seq_infodata_type)     , pointer :: infodata
+    integer                               :: rc, urc
+    type(ESMF_State)                      :: import_state, export_state
+    type(ESMF_GridComp)                   :: lnd_comp
+!-------------------------------------------------------------------------------
+
+    ! Set cdata pointers
+    call seq_cdata_setptrs(cdata, ID=COMPID, mpicom=mpicom, &
+         gsMap=gsMap, dom=dom, infodata=infodata)
+    call seq_comm_getcompstates(COMPID, lnd_comp, import_state, export_state)
+
+    ! Copy infodata to state
+    call esmfshr_infodata_infodata2state(infodata,export_state,ID=COMPID,rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    ! call into ESMF init method which calls ESMF run method
+    call ESMF_GridCompInitialize(lnd_comp, importState=import_state, exportState=export_state, clock=EClock, userRc=urc, rc=rc)
+    if (urc /= ESMF_SUCCESS) call ESMF_Finalize(rc=urc, endflag=ESMF_END_ABORT)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    ! copy export_state to infodata
+    call esmfshr_infodata_state2infodata(export_state, infodata, rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+end subroutine lnd_init_mct
+
+!===============================================================================
+!BOP ===========================================================================
+!
+! !IROUTINE: lnd_run_mct
+!
+! !DESCRIPTION:
+!     run method for the VIC lnd model
+!
+! !REVISION HISTORY:
+!
+! !INTERFACE: ------------------------------------------------------------------
+
+subroutine lnd_run_mct(EClock, cdata, x2d, d2x, cdata_s, x2s, s2x)
+
+! !INPUT/OUTPUT PARAMETERS:
+
+    type(ESMF_Clock)            ,intent(inout) :: EClock     ! clock
+    type(seq_cdata)             ,intent(inout) :: cdata
+    type(mct_aVect)             ,intent(inout) :: x2d        ! driver -> vic
+    type(mct_aVect)             ,intent(inout) :: d2x        ! vic   -> driver
+    type(seq_cdata)             ,intent(inout) :: cdata_s
+    type(mct_aVect)             ,intent(inout) :: x2s        ! driver -> vic
+    type(mct_aVect)             ,intent(inout) :: s2x        ! vic   -> driver
```

Comment:
The comments are the same, but the variables are different. Update comments to reflect the differences between `x2d` and `d2x` on the one hand and `x2s` and `s2x` on the other.